### PR TITLE
Added music list plugin to search and filter

### PR DESCRIPTION
**Pull request by achencoms**
Adds the same searching and filtering options as the QuestListPlugin to the music list.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+		{
+			currentMusicFilter = MusicState.ALL;
+			tracks = null;
+		}
+	}
+
+	@Subscribe
+	public void onVarClientIntChanged(VarClientIntChanged varClientIntChanged)
+	{
+		if (varClientIntChanged.getIndex() == VarClientInt.INVENTORY_TAB.getIndex())
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java]: Why can't this be created just when the music interface is loaded? Creating/destroying it each time the tab changes seems unnecessary.

Comment (achencoms) [reply runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java]: You're absolutely correct. I was unaware of the existence of the WidgetLoaded class prior, but I have fixed this issue.

**Comment by Adam-** [comment]
The license header needs to be added to each new file. Also @abextm can you review?

File: runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.musiclist;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.runelite.api.*;
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java]: [Fix your imports](https://github.com/runelite/runelite/wiki/Code-Conventions#imports)

File: runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+
+
+@PluginDescriptor(
+		name = "MusicListPlugin",
+		description = "Adds search and filter for the music list"
+)
+
+public class MusicListPlugin extends Plugin
+{
+
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java]: remove this newline

File: runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+	{
+		if (isChatboxOpen() && !isOnMusicTab())
+		{
+			chatboxPanelManager.close();
+		}
+	}
+
+	private boolean isOnMusicTab()
+	{
+		return client.getVar(VarClientInt.INVENTORY_TAB) == 13;
```
Comment (abextm) [inline runelite-client/src/main/java/net/runelite/client/plugins/musiclist/MusicListPlugin.java]: This should probably get an enum in runelite-api

**Pull request merged**